internal/metamorph/processor_helpers.go

```diff
@@ -1,7 +1,78 @@
 package metamorph
 
-func (p *Processor) GetProcessorMapSize() int {
+import (
+	"github.com/bitcoin-sv/arc/internal/metamorph/store"
+	"github.com/libsv/go-p2p/chaincfg/chainhash"
+)
 
+func (p *Processor) GetProcessorMapSize() int {
 	return p.ProcessorResponseMap.Len()
+}
+
+func updateStatusMap(statusUpdatesMap map[chainhash.Hash]store.UpdateStatus, statusUpdate store.UpdateStatus) {
+	foundStatusUpdate, found := statusUpdatesMap[statusUpdate.Hash]
+
+	if !found || (found && shouldUpdateStatus(statusUpdate, foundStatusUpdate)) {
+		if len(statusUpdate.CompetingTxs) > 0 {
+			statusUpdate.CompetingTxs = mergeUnique(statusUpdate.CompetingTxs, foundStatusUpdate.CompetingTxs)
+		}
+
+		statusUpdatesMap[statusUpdate.Hash] = statusUpdate
+	}
+}
+
+func shouldUpdateStatus(statusUpdate, foundStatus store.UpdateStatus) bool {
+	if statusUpdate.Status > foundStatus.Status {
+		return true
+	}
+
+	// If the statuses are both DOUBLE_SPEND_ATTEMPTED, but have
+	// different competing transactions - we want to update.
+	if !unorderedEqual(statusUpdate.CompetingTxs, foundStatus.CompetingTxs) {
+		return true
+	}
+
+	return false
+}
+
+// unorderedEqual checks if two string slices contain
+// the same elements, regardless of order
+func unorderedEqual(sliceOne, sliceTwo []string) bool {
```

Comment:
This looks correct. Alternatively maybe also https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff could be used

See https://stackoverflow.com/a/67624073

Or

```
deep.Equal([]string{"hello", "world"}, []string{"world", "hello"}, deep.FLAG_IGNORE_SLICE_ORDER) 
```

https://github.com/go-test/deep/issues/28#issuecomment-754269711

Reply:
The second option would require installing a new dependency. I'm not sure if this simple function is worth it?

As for the first option, I run some small benchmarks, which suggest that this custom implementation that I did is much faster than using a `go-cmp` with custom comparator. Here are the results, I can also include them in the code if you'd like:

```
goos: darwin
goarch: arm64
pkg: github.com/bitcoin-sv/arc/internal/metamorph
BenchmarkUnorderedEqual/txIDs-1-10              71081755                16.63 ns/op
BenchmarkUnorderedEqual/txIDs-10-10              3006540               400.3 ns/op
BenchmarkUnorderedEqual/txIDs-100-10              171099              6963 ns/op
BenchmarkUnorderedEqual/txIDs-1000-10              14553             82411 ns/op
BenchmarkUnorderedEqual/txIDs-10000-10              1507            789109 ns/op
BenchmarkCmpDiff/txIDs-1-10                       120276              9846 ns/op
BenchmarkCmpDiff/txIDs-10-10                       26148             45883 ns/op
BenchmarkCmpDiff/txIDs-100-10                       1446            820165 ns/op
BenchmarkCmpDiff/txIDs-1000-10                       177           6744339 ns/op
BenchmarkCmpDiff/txIDs-10000-10                       18          62019458 ns/op
```

Here's the function I used ad CmpDiff:
```
func cmpDiff(sliceOne, sliceTwo []string) bool {
	comparator := func(a, b string) bool { return a < b }
	return cmp.Diff(sliceOne, sliceTwo, cmpopts.SortSlices(comparator)) == ""
}
```

Reply:
Hey, nice. Then we for sure stick with the faster implementation of yours 👍